airbyte-cdk/python/airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -611,6 +614,16 @@ def create_declarative_stream(self, model: DeclarativeStreamModel, config: Confi
         stop_condition_on_cursor = (
             model.incremental_sync and hasattr(model.incremental_sync, "is_data_feed") and model.incremental_sync.is_data_feed
         )
+        client_side_incremental_sync = None
+        if (
+            model.incremental_sync
+            and hasattr(model.incremental_sync, "is_client_side_incremental")
+            and model.incremental_sync.is_client_side_incremental
+        ):
+            client_side_incremental_sync = {
+                "date_time_based_cursor": self._create_component_from_model(model=model.incremental_sync, config=config),
+                "per_partition_cursor": combined_slicers if isinstance(combined_slicers, PerPartitionCursor) else None,
```

Comment:
should we fail if combined_slicer is set but isn't a `PerPartitionCursor`? That would be unexpected since `_merge_stream_slicers` returns a `PerPartitionCursor` if there's both a stream slicer and an `incremental_sync`. I'd rather fail early than troubleshoot unexpected and undefined behavior in 6 months

Reply:
added exception